### ISSUE TITLE
Chips: fails when used with CommandButton

### ISSUE THREAD

**Issue by ybendek** [primefaces/primefaces#9832]
### Describe the bug

The pretty basic behaviour of chips component is failing when in the same page you have a p:commandButton component. Normally as it's displayed in the pf showcase if you press the "enter" key in your keyboard a new chip (tag) is added to the list. However if you have a p:commandButton in the same page the following error is displayed in the console

`Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at d.addItem (components.js?ln=primefaces&v=12.0.3:965:48)
    at HTMLInputElement.<anonymous> (components.js?ln=primefaces&v=12.0.3:964:22)
    at HTMLInputElement.dispatch (jquery.js?ln=primefaces&v=12.0.3:2:43064)
    at y.handle (jquery.js?ln=primefaces&v=12.0.3:2:41048)`

If you replace the p:commandButton for a h:commandButton then it works as expected, but obviously this is not the desired behaviour.

### Reproducer

In the reproducer attached, you will find a bean an a XHTML file with the example.

1) Call the test page
2) add some characters in the component
3) press "enter" in your keyboard
4) then the error is displayed in the browser console

If you change the p:commandButton for a h:commandButton and repeat the steps, it works.

It's important to mention that I was using before PF 11.0.9 and it works with that, but now with PF 12.0.3 it fails.

[reproducerIssue.zip](https://github.com/primefaces/primefaces/files/10830154/reproducerIssue.zip)

### Expected behavior

Every time you press "enter" a new element should be added to the component

### PrimeFaces edition

Elite

### PrimeFaces version

12.0.3

### Theme

Rain

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

11

**Comment by melloware** [comment]
I will take a look. Chips ENTER key should be stopping the ENTER key from bubbling up.

**Comment by melloware** [comment]
I can't reproduce.  Attached is a runnable reproducer:
[pf-9832.zip](https://github.com/primefaces/primefaces/files/10831038/pf-9832.zip)

Run `mvn clean jetty:run` and navigate to http://localhost:8080/primefaces-test/

**Comment by melloware** [comment]
Here is a MonkeyPatch you can use to fix 12.0.3 for now.

```js
if (PrimeFaces.widget.Chips) {
    PrimeFaces.widget.Chips.prototype.addItem = function(value, refocus) {
        var $this = this;

        if (!value || !value.trim().length) {
            return;
        }

        var tokens = value.split(this.cfg.separator);
        for (var i = 0; i < tokens.length; i++) {
            var token = tokens[i];
            if (token && token.trim().length && (!this.cfg.max || this.cfg.max > this.hinput.children('option').length)) {
                var escapedValue = PrimeFaces.escapeHTML(token);

                if (this.cfg.unique) {
                    var duplicateFound = false;
                    this.hinput.children('option').each(function() {
                        if (this.value === escapedValue) {
                            duplicateFound = true;
                            return false; // breaks
                        }
                    });
                    if (duplicateFound) {
                        $this.refocus(refocus);
                        continue;
                    }
                }

                var itemDisplayMarkup = '<li class="ui-chips-token ui-state-active ui-corner-all">';
                itemDisplayMarkup += '<span class="ui-chips-token-icon ui-icon ui-icon-close"></span>';
                itemDisplayMarkup += '<span class="ui-chips-token-label">' + escapedValue + '</span></li>';

                this.inputContainer.before(itemDisplayMarkup);
                this.refocus(refocus);

                this.hinput.append('<option value="' + escapedValue + '" selected="selected"></option>');
                this.invokeItemSelectBehavior(escapedValue);
            }
        }
        $this.updateFloatLabel();
    };
};
```

**Comment by ybendek** [comment]
> 

Awsome, thank you, this patch solve the issue while we wait for 12.0.4.
